pandas/core/generic.py

```diff
@@ -9870,13 +9989,14 @@ def abs(self: FrameOrSeries) -> FrameOrSeries:
         """
         return np.abs(self)
 
+    @final
     def describe(
         self: FrameOrSeries,
         percentiles=None,
         include=None,
         exclude=None,
         datetime_is_numeric=False,
-    ) -> FrameOrSeries:
+    ) -> FrameOrSeriesUnion:
```

Comment:
this shouldn't change. AFAICT Series.describe() always returns a Series and DataFrame.describe() always returns a DataFrame.

mypy cannot narrow types with`if self.ndim == 1` on L10348.

since self is a typevar, can't use cast either.

so could either
1. ignore the error. we know better than mypy
2. use `isinstance(self.ndim, Series)`
3. have seperate DataFrame and Series methods 

1. is probably the easiest.

Reply:
OK to leave as is for now, since a subclassed Series returns a Series, so the typevar restriction falls over.

This is part of the public API, so the status quo would be better for regular users though.

Reply:
ok, will go with 1